stdlib/event/time.lua

```diff
@@ -0,0 +1,73 @@
+--- Time Event module
+-- @module Event.Time
+
+require 'stdlib/event/event'
+
+Event.Time = {}
+Event.Time._last_change = {}
+
+--All times are offset by 0.5
+--This is because both EvoGUI and MoWeather already apply that offset.
+--Following the precedent to remain consistent.
+--Actually, this little snippet is even borrowed from EvoGUI.
+if remote.interfaces.MoWeather then
+    -- assume MoWeather's getdaytime is sane
+    function Event.Time.get_day_time(surface_name_or_index) return remote.call("MoWeather", "getdaytime", surface_name_or_index) end
+else
+    -- 0.5 is midnight; let's make days *start* at midnight instead.
+    function Event.Time.get_day_time(surface_name_or_index) return game.surfaces[surface_name_or_index].daytime + 0.5 end
+end
+
+--These are Surface only events
+--- @field Gets around ldoc V1.4.3 eating the first field. :(
+Event.Time.ldoc_V1_4_3_eats_this   = nil
+--- @field Fires whenever it becomes midday/noon on a surface
+Event.Time.midday   = script.generate_event_name()
+--- @field Fires whenever it becomes midnight on a surface
+Event.Time.midnight = script.generate_event_name()
+--- @field Fires whenever the sunrises on a surface
+Event.Time.sunrise  = script.generate_event_name()
+--- @field Fires whenever the sunsets on a surface
+Event.Time.sunset   = script.generate_event_name()
+--- @field Fires every hour for a surface
+Event.Time.hourly   = script.generate_event_name()
+--- @field Fires every day for a surface
+Event.Time.daily    = script.generate_event_name()
+
+Event.register(defines.events.on_tick, function(event)
+    for _, surface in pairs(game.surfaces) do
+        local day_time = math.fmod(Event.Time.get_day_time(surface.index), 1)
+        local day_time_minutes = math.floor(day_time * 24 * 60)
+
+        if day_time_minutes ~= Event.Time._last_change[surface.name] then
+            Event.Time._last_change[surface.name] = day_time_minutes
+
+            if day_time_minutes % 60 == 0 then
+                game.raise_event(Event.Time.hourly, {surface = surface})
```

Comment:
Do we actually want game.raise_event here...

Say 2 mods use stdlib for the time events. Won't they raise these both raise events......

However, I think they time events will have different event names because we aren't exposing the event ids to the remote interface....

So this is technically okay, but duplicated logic in each mod then...

So back to my point. Not sure we actually need to do a raise_event if this event is only ever seen by the "mod" that stdlib is in. Event.dispatch would work and not hurt performance by hitting the factorio API.